RestCommClient/Classes/Media/XirsysTURNClient.m

```diff
@@ -34,109 +34,74 @@ @interface XirsysTURNClient ()
 
 @implementation XirsysTURNClient
 
-- (instancetype)initWithURL:(NSURL *)url {
+- (instancetype)initWithURL:(NSURL *)url{
     NSParameterAssert([url absoluteString].length);
     if (self = [super init]) {
         _url = url;
     }
     return self;
 }
 
+- (void)requestServersWithUsername:(NSString *)username password:(NSString *)password andCompletionHandler:(void (^)(NSArray *turnServers, NSError *error))completionHandler
+{
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:_url];
+    NSString *basic = [NSString stringWithFormat:@"%@:%@",username, password];
+    NSData *encodeData = [basic dataUsingEncoding:NSUTF8StringEncoding];
+    NSString *authValue = [NSString stringWithFormat:@"Basic %@", [encodeData base64EncodedStringWithOptions:0]];
+    [request setValue:authValue forHTTPHeaderField:@"Authorization"];
+    [request setHTTPMethod:@"PUT"];
+   
+    [self requstServersWithRequest:request andCompletionHandler:completionHandler];
+    
+}
+
 - (void)requestServersWithCompletionHandler:(void (^)(NSArray *turnServers, NSError *error))completionHandler
 {
-    //NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:_url];
-    // We need to set origin because TURN provider whitelists requests based on
-    // origin.
-    //[request addValue:@"Mozilla/5.0" forHTTPHeaderField:@"user-agent"];
-    //[request addValue:kTURNOriginURLString forHTTPHeaderField:@"origin"];
+    NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:_url];
+    [self requstServersWithRequest:request andCompletionHandler:completionHandler];
+}
+
+- (void)requstServersWithRequest:(NSMutableURLRequest *)request andCompletionHandler:(void (^)(NSArray *turnServers, NSError *error))completionHandler{
```

Comment:
Small typo requstServersWithRequest

Reply:
Nice that you can use the same facilities for both V2 and V3. That's what I meant when we discussed this

Reply:
done.